services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx

```diff
@@ -296,163 +391,301 @@ export default class ActiveClassrooms extends React.Component<ActiveClassroomsPr
-      />)
+      return (
+        <RenameClassModal
+          classroom={selectedClassroom}
+          close={this.closeModal}
+          onSuccess={this.onSuccess}
+        />
+      )
     }
   }
 
   renderChangeGradeModal() {
     const { showModal, classrooms, selectedClassroomId } = this.state
     const selectedClassroom = classrooms.find(c => c.id === selectedClassroomId)
     if (showModal === changeGradeModal && selectedClassroom) {
-      return (<ChangeGradeModal
-        classroom={selectedClassroom}
-        close={this.closeModal}
-        onSuccess={this.onSuccess}
-      />)
+      return (
+        <ChangeGradeModal
+          classroom={selectedClassroom}
+          close={this.closeModal}
+          onSuccess={this.onSuccess}
+        />
+      )
     }
   }
 
   renderArchiveClassModal() {
     const { showModal, classrooms, selectedClassroomId } = this.state
     const selectedClassroom = classrooms.find(c => c.id === selectedClassroomId)
     if (showModal === archiveClassModal && selectedClassroom) {
-      return (<ArchiveClassModal
-        classroom={selectedClassroom}
-        close={this.closeModal}
-        onSuccess={this.onSuccess}
-      />)
+      return (
+        <ArchiveClassModal
+          classroom={selectedClassroom}
+          close={this.closeModal}
+          onSuccess={this.onSuccess}
+        />
+      )
+    }
+  }
+
+  renderImportCleverClassroomStudentsModal() {
+    const { showModal, classrooms, selectedClassroomId } = this.state
+    const selectedClassroom = classrooms.find(c => c.id === selectedClassroomId)
+
+    if (showModal === importCleverClassroomStudentsModal && selectedClassroom) {
+      return (
+        <ImportCleverClassroomStudentsModal
+          classroom={selectedClassroom}
+          close={this.closeModal}
+          onSuccess={this.onSuccess}
+        />
+      )
     }
   }
 
   renderImportGoogleClassroomStudentsModal() {
-    const { user, } = this.props
     const { showModal, classrooms, selectedClassroomId } = this.state
     const selectedClassroom = classrooms.find(c => c.id === selectedClassroomId)
     if (showModal === importGoogleClassroomStudentsModal && selectedClassroom) {
-      return (<ImportGoogleClassroomStudentsModal
-        classroom={selectedClassroom}
-        close={this.closeModal}
-        onSuccess={this.onSuccess}
-      />)
+      return (
+        <ImportGoogleClassroomStudentsModal
+          classroom={selectedClassroom}
+          close={this.closeModal}
+          onSuccess={this.onSuccess}
+        />
+      )
+    }
+  }
+
+  renderImportCleverClassroomsModal() {
+    const { user } = this.props
+    const { cleverClassrooms, showModal } = this.state
+
+    if (showModal === importCleverClassroomsModal) {
+      return (
+        <ImportCleverClassroomsModal
+          classrooms={cleverClassrooms}
+          close={() => this.closeModal(() => this.setState({attemptedImportCleverClassrooms: false}))}
```

Comment:
We try to avoid binding in props in general (https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md), but there is an existing pattern of doing it in this file since it was written before we codified our JS style guide so I think it's okay to follow that in this instance. I think we should definitely define the argument you're passing to closeModal as its own function, though, so we aren't passing `two` arrow functions as a prop.